### PR TITLE
Multiple improvements

### PR DESCRIPTION
**Pull request by Charles-ErickBG**
Sorry guys for the size of the PR. This is my fist time and I promise I will try my best to never do that to you again :| .

Please, focus more on the **code** and **structure** than on contents, since the idea was more to see how we could integrate the doc styleguide in Vapor than anything else. I wanted to see how we could add new sections, new sections/components and integrate what's already in the styleguide. 

I'm aware that there will probably be a lot of stuff for me to adjust. I am ready to learn, masters!

Note that Vapor documentation improvement should be very iterative, so this is the first of a series of many (smaller than this one, I swear).

Details
- reorder entries in the Components section in alphabetical order
- add General Guidelines section and content
- add Buttons Overview section/component
- improve components styles for better readability
- change header styles to better match that of the usual Coveo products, such as the Admin

### CODE REVIEW

**Comment by Charles-ErickBG** [comment]
Capture d’écran, le 2019-04-02 à 13 03 18
Capture d’écran, le 2019-04-02 à 13 04 30
Capture d’écran, le 2019-04-02 à 13 04 46

**Comment by otaillon** [comment]
> * Not sure I like to have a `/`in the name of a section. I feel like those should construct a section themselves. For example all `Card / Something` could be children of the `Card` section.

Do you know of a way to have a third-level nested nav ?

**Comment by Charles-ErickBG** [comment]
I agree with @gdostie. 
1. I don't like the / either. I planned to take those out in separate sections in a next PR. I don't know what I'll do with the GENERAL GUIDELINES section though, since it's gonna be hard to keep it clean and simple if we split the general guidelines in separate sections as well. I might keep the / in this one but do what needs to be done to remove them from the components section.
2. I was aware that we usually don't push commented code on master :). I just didn't want to loose what I had done and my PR was already pretty big, wasn't it xD. I'll copy it on my side and remove it from the code.

**Comment by gdostie** [comment]
@otaillon We have none implemented as of now, but I think it could be useful in the present use case. However, I don't know if it would be a good practice UX wise. Perhaps you could enlighten us on that matter :bulb: In any case, we could possibly use a tab navigation to browse through the different card types inside the `Card` section. Just thinking out loud here, it is probably out of the scope of the current PR.

**Comment by Charles-ErickBG** [comment]
Capture d’écran, le 2019-04-03 à 16 51 00

File: docs/_components/general_guidelines/01_content_numbers.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+layout: default
+type: generalGuidelines
+title: Content / Numbers
+source_visibility: hidden
+---
+<div class="spaced-boxes-container flex flex-wrap">
+    <p class="h3">Always use a comma to separate thousand, e.g. 23, 000, 000.</p>
```
Comment (GermainBergeron) [inline docs/_components/general_guidelines/01_content_numbers.html]: I think we don't put space after the comma

File: docs/_includes/component.html [hunk 2]
```diff
@@ -7,10 +8,10 @@
             {{ entry.content }}
         </div>
         <div class="clearfix"></div>
-        <div class="sg-component-source">
+        <div class="sg-component-source {{ entry.source_visibility }}">
             {% highlight html %}{{ entry.content }}{% endhighlight %}
         </div>
-        <form class="codepen-form" action="http://codepen.io/pen/define" method="POST" target="_blank">
+        <form class="codepen-form" action="http://codepen.io/pen/define" method="POST" target="_blank" hidden="{{ entry.nosource }}">
```
Comment (GermainBergeron) [inline docs/_includes/component.html]: Is this change still needed?

Comment (Charles-ErickBG) [reply docs/_includes/component.html]: Je vais l'arranger. Ça ne devrait pas être entry.nosource dans un hidden=, mais plutôt comme le reste entry.source_visibility en tant que classe sur l'élément.

L'idée est de cacher le bouton "CODEPEN" sur les éléments de documentation.

File: docs/_components/buttons/01_button-overview.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        <li><p class="h3">Button <span class="bold">labels</span> should be <span class="bold">short</span> and use <span class="bold">“trigger words”</span>.</p></li>
+        <li><p class="h3"><span class="bold">Avoid</span> using the terms <span class="bold">OK</span>, <span class="bold">Yes</span> and <span class="bold">No</span> to dismiss a Dialog Modal.</p></li>
+        <li><p class="h3">As much as possible, the <span class="bold">first word</span>of the label should be an <span class="bold">action verb</span> (e.g., the label for a button that allows the user to add a notification should be “ADD NOTIFICATION”).</p></li>
+        <li><p class="h3">When a button includes <span class="bold">only an icon</span> (no label), the label should be added in the <span class="bold">alternative text</span>element in the HTML and appear in a <span class="bold">tooltip</span>when the cursor is above the button.</p></li>
+        <li><p class="h3">An <span class="bold">icon with a label does not need alternative text</span>. In this case, the icon is a visual help to identify the purpose faster and filling it will cause a screen reader to repeat the text twice.</p></li>
```
[the remark this answers is not in the document]
Comment (Charles-ErickBG) [reply docs/_components/buttons/01_button-overview.html]: Not sure either. I haven't written those :).

But it should be fine like this, I believe. Thanks.
